File: NCore/Commands/Commandcraft.cs
using System;
using System.Threading.Tasks;

namespace NCore
{
    public class Commandcraft : Command
    {
        public Commandcraft() : base("craft", "Craft item [deprecated; use craft menu instead]", "craft <material>")
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            if (!sender.IsPlayer)
            {
                return false;
            }

            if (args.Length == 1)
            {
                try
                {
                    var m = Enum.Parse(typeof(Material), args[0].ToUpper());
                    NetworkPlayer p = (NetworkPlayer)sender;
                    await p.Craft((Material)m);
                    await p.Chat("Crafted!");
                }
                catch (Exception ex)
                {
                    await sender.SendMessage("An internal error occured.");
                }

                return true;
            }

            return false;
        }
    }
}

File: NetcraftServer/Commands/Commandcraft.vb
Imports NetcraftServer

Public Class Commandcraft
    Inherits Command
    Sub New()
        MyBase.New("craft", "Скрафтить вещь [ИГРОК]", "craft <material>")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.IsPlayer Then
            Return False
        End If
        If args.Length = 1 Then
            Try
                Dim m = [Enum].Parse(GetType(Material), args(0).ToUpper)
                Dim p As NetworkPlayer = sender
                p.Craft(m)
                p.Chat("Crafted!")
            Catch ex As Exception
                sender.SendMessage("An internal error occured.")
            End Try
            Return True
        End If
        Return False
    End Function
End Class
